R/sensorreading-data.R

```diff
@@ -0,0 +1,231 @@
+      
+      if(!is.null(listElements$estimatedTime) && !is.na(listElements$estimatedTime)) {
+        estDateTime <- (strsplit(listElements$estimatedTime, split="[T]"))
+        estDate <- strftime(estDateTime[[1]][1], "%m/%d/%Y")
+      } else {
+        estDate <- ""
+      }
+      
+      #Get the time out of the nearest corrected iv time, don't need the date
+      if (!is.null(listElements$nearestrawTime) || is.na(listElements$nearestrawTime)) {
+        dateTimeRaw <- (strsplit(listElements$nearestrawTime, split="[T]"))
+        dateRaw <- strftime(dateTimeRaw[[1]][1], "%m/%d/%Y")
+        
+        #Break apart, format dates/times, put back together.
+        timeFormattingRaw <- sapply(dateTimeRaw[[1]][2], function(s) strsplit(s,split="[-]")[[1]])
+        timeFormattingRaw[[1]] <- sapply(timeFormattingRaw[[1]], function(s) sub(".000","",s))
+        timeFormattingRaw[[2]] <- paste(" (UTC",timeFormattingRaw[[2]], ")")
+        timeFormattingRaw <-  paste(timeFormattingRaw[[1]],timeFormattingRaw[[2]]) 
+      }
+    }
+    
+    rec <- getRecorderWithinUncertainty(listElements$recorderUncertainty, listElements$value, listElements$recorderValue)
+    ind <- getIndicatedCorrection(listElements$recorderValue, listElements$value)
+    app <- getAppliedCorrection(listElements$nearestrawValue, listElements$nearestcorrectedValue)
+    corr <- getCorrectedRef(listElements$value, listElements$nearestcorrectedValue, listElements$uncertainty)
+    
+    toAdd = c(date,
+              timeFormatting,
+              nullMask(listElements$party), 
+              nullMask(listElements$sublocation),
+              ##
+              nullMask(listElements$monitoringMethod),
+              nullMask(listElements$type),
+              nullMask(listElements$value),
+              nullMask(listElements$uncertainty),
+              ##
+              nullMask(listElements$recorderMethod),
+              nullMask(listElements$recorderType),
+              nullMask(listElements$recorderValue),
+              nullMask(listElements$recorderUncertainty), 
+              ##
+              rec, 
+              ind, 
+              app, 
+              corr,
+              ##
+              nullMask(listElements$nearestrawValue),
+              timeFormattingRaw
+              #qualifiers?
+              )
+    
+    toRet <- rbind(toRet, data.frame(t(toAdd),stringsAsFactors = FALSE))
+  }
+  colnames(toRet) <- columnNames
+  rownames(toRet) <- NULL
+  
+  return(toRet)
+}
+
+nullMask <- function(val) {
+  if(!is.null(val)) {
+    result <- val
+  } else {
+    result <- ""
+  }
+  return(result)
+}
+
+getQualifiers <- function(time, inQualifiers) {
+  if(length(inQualifiers) < 1) return("");
+  q <- inQualifiers[[1]]
+  
+  if(is.null(q) || length(q) < 1) return("");
+  
+  qualifiers <- q[time>q$startDate & q$endDate>time,]
+  
+  builtQualifiers <- ""
+  if(nrow(qualifiers) > 0) {
+    for(i in 1:nrow(qualifiers)) {
```

Comment:
I think instead of having both the if and for statement, you can just use `for(i in seq_len(nrow(qualifiers)))`. It will automatically not do anything in the loop since there is nothing to loop through. 

```r
qualifiers <- data.frame(1:5)
qualifiers2 <- data.frame()

for(i in seq_len(nrow(qualifiers))){
  print("stuff")
}

for(i in seq_len(nrow(qualifiers2))){
  print("stuff")
}
```